mito-ai/src/components/ModelSelector.tsx

```diff
@@ -0,0 +1,221 @@
+const MAX_MODEL_NAME_LENGTH = 18;
+
+interface ModelSelectorProps {
+  onConfigChange: (config: ModelConfig) => void;
+}
+
+const ModelSelector: React.FC<ModelSelectorProps> = ({ onConfigChange }) => {
+  const [selectedModel, setSelectedModel] = useState<string>('gpt-4o-mini');
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [dropdownWidth, setDropdownWidth] = useState<number>(0);
+  const [isCompact, setIsCompact] = useState<boolean>(false);
+  const selectedModelRef = useRef<HTMLSpanElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  // Load config from localStorage on component mount and notify parent
+  useEffect(() => {
+    const storedConfig = localStorage.getItem('llmModelConfig');
+    if (storedConfig) {
+      try {
+        const parsedConfig = JSON.parse(storedConfig);
+        const model = parsedConfig.model || 'gpt-4o-mini';
+        setSelectedModel(model);
+
+        onConfigChange({ model });
+
+      } catch (e) {
+        console.error('Failed to parse stored LLM config', e);
+      }
+    }
+  }, [onConfigChange]);
+
+  // Set up resize observer to detect container width changes
+  useEffect(() => {
+    const observer = new ResizeObserver(entries => {
+      for (const entry of entries) {
+        const containerWidth = entry.contentRect.width;
+        setIsCompact(containerWidth < 300);
+      }
+    });
+
+    const chatControls = containerRef.current?.closest('.chat-controls');
+    if (chatControls) {
+      observer.observe(chatControls);
+    }
+
+    return () => observer.disconnect();
+  }, []);
+
+  useEffect(() => {
+    if (selectedModelRef.current && !isCompact) {
+      const textWidth = selectedModelRef.current.scrollWidth;
+      setDropdownWidth(Math.min(textWidth + 30, 150)); // Set a maximum width of 150px
+    } else if (isCompact) {setDropdownWidth(28);
+    }
+  }, [selectedModel, isCompact]);
+
+  // Shorten model name if it's too long
+  const truncateModelName = (name: string): string => {
+    if (name.length > MAX_MODEL_NAME_LENGTH) {
+      return `${name.substring(0, MAX_MODEL_NAME_LENGTH - 3)}...`;
+    }
+    return name;
+  };
+
+  const handleModelChange = (model: string) => {
+    setSelectedModel(model);
+    setIsOpen(false);
+
+    const newConfig = {
+      model: model
+    };
+
+    localStorage.setItem('llmModelConfig', JSON.stringify(newConfig));
+
+    // Notify parent component
+    onConfigChange(newConfig);
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const target = event.target as HTMLElement;
+      if (!target.closest('.model-selector')) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
```

Comment:
It looks like all of this code is not really helpful, if I'm understanding it correctly. 

1. The first section of code is checking if the dropdown should be above or below? As far as I can tell there's no case where the dropdown should be below. 
2. The second chunk of code figures out where the the dropdown should be located horizontally? But even when I delete all of this code it seems positioned perfectly fine. 

I think if we delete this code this entire useEffect.